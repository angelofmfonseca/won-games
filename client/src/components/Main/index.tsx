import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo src="/img/logo.svg" alt="logo" />
    <S.Title>Won Games</S.Title>
    <S.Description>
      A e-commerce project for games build with Typescript, React, Next and
      Styled-Components
    </S.Description>
    <S.Illustration src="/img/hero-illustration.svg" alt="main image" />
  </S.Wrapper>
)

export default Main
