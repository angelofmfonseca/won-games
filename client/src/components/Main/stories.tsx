import { withKnobs, text } from '@storybook/addon-knobs'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
}

export const Basic = () => (
  <Main
    title={text('Title', 'Won Games')}
    description={text(
      'Description',
      'An e-commerce project for games build with Typescript, React, Next and Styled-Components'
    )}
  />
)
