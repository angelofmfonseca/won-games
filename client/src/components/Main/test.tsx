import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('sould render heading', () => {
    const { container } = render(<Main />)
    expect(
      screen.getByRole('heading', { name: /Won Games/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
