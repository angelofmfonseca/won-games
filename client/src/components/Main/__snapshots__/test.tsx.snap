// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> sould render heading 1`] = `
<main
  class="styles__Wrapper-czREMl bljgDb"
>
  <img
    alt="logo"
    class="styles__Logo-hpIqnx hQjETQ"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-fMTgIO fjrrRF"
  >
    Won Games
  </h1>
  <h2
    class="styles__Description-htEhDc kiZXgT"
  >
    A e-commerce project for games build with Typescript, React, Next and Styled-Components
  </h2>
  <img
    alt="main image"
    class="styles__Illustration-fhoxhO hNVvPh"
    src="/img/hero-illustration.svg"
  />
</main>
`
